### PR TITLE
Change color of overwritten settings equal to overwrite to orange

### PR DESCRIPTION
**Pull request by veteran29**
**When merged this pull request will:**
- title

image

### CODE REVIEW

**Comment by commy2** [comment]
>with some combinations I was able to achieve green cross icon, not sure why

lol

**Comment by veteran29** [comment]
Ready for review

File: addons/settings/stringtable.xml [hunk 1]
```diff
@@ -406,12 +406,16 @@
...
-            <Polish>Ta wartość zostałą nadpisana przez klientów i ma zastosowanie tylko na serwerze.</Polish>
+            <Polish>Ta wartość została nadpisana przez klientów i ma zastosowanie tylko na serwerze.</Polish>
             <French>Ce paramètre est écrasé par les clients et ne s'applique qu'au serveur.</French>
             <Turkish>Oyuncular tarafından bu ayarın üstüne yazılmış, ayar sadece sunucuda geçerli.</Turkish>
             <Italian>Questa impostazione è sovrascritta dai client e si applica solo al server.</Italian>
             <Czech>Toto nastavení je přepsáno klienty a platí pouze na serveru.</Czech>
         </Key>
+        <Key ID="STR_cba_settings_overwritten_equal">
+            <English>\nOverwriting value is equal.</English>
+            <Polish>\nNadpisująca wartość jest identyczna.</Polish>
```
Comment (Elgin675) [inline addons/settings/stringtable.xml]:
```suggestion
            <Polish>\nNadpisująca wartość jest identyczna.</Polish>
            <French>\nLa valeur de remplacement est la même.</French>
```

Comment (commy2) [reply addons/settings/stringtable.xml]: We decided to change this string. Will happen soonTM.

File: addons/settings/stringtable.xml [hunk 2]
```diff
@@ -406,12 +406,27 @@
...
+            <Polish>Ta wartość została nadpisana przez klientów i ma zastosowanie tylko na serwerze.</Polish>
             <French>Ce paramètre est écrasé par les clients et ne s'applique qu'au serveur.</French>
             <Turkish>Oyuncular tarafından bu ayarın üstüne yazılmış, ayar sadece sunucuda geçerli.</Turkish>
             <Italian>Questa impostazione è sovrascritta dai client e si applica solo al server.</Italian>
             <Czech>Toto nastavení je přepsáno klienty a platí pouze na serveru.</Czech>
         </Key>
+        <Key ID="STR_cba_settings_overwritten_by_client_equal_tooltip">
+            <English>Overwritten by clients with the same value.</English>
+            <German>Mit gleichem Wert durch Client überschrieben.</German>
+            <Polish>Nadpisane przez klientów z taką samą wartością.</Polish>
```
Comment (Elgin675) [inline addons/settings/stringtable.xml]:
```suggestion
            <Polish>Nadpisane przez klientów z taką samą wartością.</Polish>
            <French>Écrasé par les clients, avec la même valeur.</French>
```

File: addons/settings/stringtable.xml [hunk 3]
```diff
@@ -406,12 +406,27 @@
...
+        </Key>
+        <Key ID="STR_cba_settings_overwritten_by_mission_equal_tooltip">
+            <English>Overwritten by mission with the same value.</English>
+            <German>Mit gleichem Wert durch Mission überschrieben.</German>
+            <Polish>Nadpisane przez misję z taką samą wartością.</Polish>
```
Comment (Elgin675) [inline addons/settings/stringtable.xml]:
```suggestion
            <Polish>Nadpisane przez misję z taką samą wartością.</Polish>
            <French>Écrasé par la mission, avec la même valeur.</French>
```

File: addons/settings/stringtable.xml [hunk 4]
```diff
@@ -406,12 +406,27 @@
...
+        </Key>
+        <Key ID="STR_cba_settings_overwritten_by_server_equal_tooltip">
+            <English>Overwritten by server with the same value.</English>
+            <German>Mit gleichem Wert durch Server überschrieben.</German>
+            <Polish>Nadpisane przez serwer z taką samą wartością.</Polish>
```
Comment (Elgin675) [inline addons/settings/stringtable.xml]:
```suggestion
            <Polish>Nadpisane przez serwer z taką samą wartością.</Polish>
            <French>Écrasé par le serveur, avec la même valeur.</French>
```

File: addons/settings/script_component.hpp [hunk 5]
```diff
@@ -138,6 +139,12 @@
     (uiNamespace getVariable QGVAR(serverTemp))  getVariable [setting, [[setting,  "server"] call FUNC(get), nil]] select 0\
 ] select (["client", "mission", "server"] find TEMP_PRIORITY(setting)))
 
+#define SOURCE_VALUE(setting,source) ([\
```
Comment (commy2) [inline addons/settings/script_component.hpp]: TEMP_VALUE_SOURCE would be a more accurate name.
Also, the macro above could then be simplified:
```cpp
#define TEMP_VALUE(setting) TEMP_VALUE_SOURCE(setting,TEMP_PRIORITY(setting))
```

File: addons/settings/stringtable.xml [hunk 6]
```diff
@@ -406,12 +406,30 @@
...
             <Russian>Этот параметр перезаписан клиентом и применяется исключительно к серверу.</Russian>
-            <Polish>Ta wartość zostałą nadpisana przez klientów i ma zastosowanie tylko na serwerze.</Polish>
+            <Polish>Ta wartość została nadpisana przez klientów i ma zastosowanie tylko na serwerze.</Polish>
             <French>Ce paramètre est écrasé par les clients et ne s'applique qu'au serveur.</French>
             <Turkish>Oyuncular tarafından bu ayarın üstüne yazılmış, ayar sadece sunucuda geçerli.</Turkish>
             <Italian>Questa impostazione è sovrascritta dai client e si applica solo al server.</Italian>
             <Czech>Toto nastavení je přepsáno klienty a platí pouze na serveru.</Czech>
         </Key>
+        <Key ID="STR_cba_settings_overwritten_by_client_equal_tooltip">
+            <English>Overwritten by clients with the same value.</English>
```
Comment (commy2) [inline addons/settings/stringtable.xml]: You don't know that. You only know about your client (singular, which is why I told you to change to singular in German too).

**Comment by commy2** [comment]
Looks very clean, so the settings stuff can't be written all that bad, right??

**Comment by veteran29** [comment]
Never said it was, it just feels little bit different from other more CBA Event based stuff. But it's fairly easy to work on it after you learn what is where ;P

**Comment by commy2** [comment]
Sadly events don't really work outside of missions, and this stuff is also meant for the editor and main menu.

File: addons/settings/stringtable.xml [hunk 7]
```diff
@@ -406,12 +406,30 @@
...
+            <German>Mit gleichem Wert durch Client überschrieben.</German>
+            <Polish>Nadpisane przez klientów z taką samą wartością.</Polish>
+            <French>Écrasé par les clients, avec la même valeur.</French>
```
Comment (Elgin675) [inline addons/settings/stringtable.xml]:
```suggestion
            <French>Écrasé par le client, avec la même valeur.</French>
```

**Comment by veteran29** [comment]
Applied changes from suggestions

File: addons/settings/script_component.hpp [hunk 8]
```diff
@@ -138,6 +139,14 @@
     (uiNamespace getVariable QGVAR(serverTemp))  getVariable [setting, [[setting,  "server"] call FUNC(get), nil]] select 0\
 ] select (["client", "mission", "server"] find TEMP_PRIORITY(setting)))
 
+#define TEMP_VALUE_SOURCE(setting,source) ([\
+    (uiNamespace getVariable QGVAR(clientTemp))  getVariable [setting, [[setting,  "client"] call FUNC(get), nil]] select 0,\
+    (uiNamespace getVariable QGVAR(missionTemp)) getVariable [setting, [[setting, "mission"] call FUNC(get), nil]] select 0,\
+    (uiNamespace getVariable QGVAR(serverTemp))  getVariable [setting, [[setting,  "server"] call FUNC(get), nil]] select 0\
+] select (["client", "mission", "server"] find source))
+
+#define TEMP_VALUE(setting) TEMP_VALUE_SOURCE(setting,TEMP_PRIORITY(setting))
```
Comment (commy2) [inline addons/settings/script_component.hpp]: Now we have TEMP_VALUE() twice.

File: addons/settings/script_component.hpp [hunk 9]
```diff
@@ -138,6 +139,14 @@
     (uiNamespace getVariable QGVAR(serverTemp))  getVariable [setting, [[setting,  "server"] call FUNC(get), nil]] select 0\
 ] select (["client", "mission", "server"] find TEMP_PRIORITY(setting)))
 
+#define TEMP_VALUE_SOURCE(setting,source) ([\
+    (uiNamespace getVariable QGVAR(clientTemp))  getVariable [setting, [[setting,  "client"] call FUNC(get), nil]] select 0,\
+    (uiNamespace getVariable QGVAR(missionTemp)) getVariable [setting, [[setting, "mission"] call FUNC(get), nil]] select 0,\
+    (uiNamespace getVariable QGVAR(serverTemp))  getVariable [setting, [[setting,  "server"] call FUNC(get), nil]] select 0\
+] select (["client", "mission", "server"] find source))
```
Comment (commy2) [inline addons/settings/script_component.hpp]: Should probably put source into parenthesis.

File: addons/settings/script_component.hpp (hunk 8 above)
Comment (veteran29) [reply addons/settings/script_component.hpp]: Huh? Somehow I've not staged removal of the old one.

Comment (veteran29) [reply addons/settings/script_component.hpp]: Fixed

**Pull request merged**